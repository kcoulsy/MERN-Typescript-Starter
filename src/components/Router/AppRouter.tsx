import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

import Home from '../../pages/Home';
import Register from '../../pages/Register';
import Login from '../../pages/Login';
import Secret from '../../pages/Secret';

import PrivateRoute from './PrivateRoute';
import PublicRoute from './PublicRoute';

const AppRouter: React.FunctionComponent = ({ children }) => {
    return (
        <Router>
            {children}
            <Switch>
                <Route exact path="/">
                    <Home />
                </Route>
                <PublicRoute path="/register">
                    <Register />
                </PublicRoute>
                <PublicRoute path="/login">
                    <Login />
                </PublicRoute>
                <PrivateRoute path="/secret">
                    <Secret />
                </PrivateRoute>
            </Switch>
        </Router>
    );
}

export default AppRouter;
