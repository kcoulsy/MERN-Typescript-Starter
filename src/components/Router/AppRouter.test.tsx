import React from 'react';
import { shallow } from 'enzyme';

import { shallowToJson } from 'enzyme-to-json';
import AppRouter from './AppRouter';

describe('render AppRouter component', () => {
    it('should render AppRouter component', () => {
        const wrapper = shallow(<AppRouter />);

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});
