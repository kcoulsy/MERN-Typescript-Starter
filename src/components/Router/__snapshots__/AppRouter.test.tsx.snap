// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render AppRouter component should render AppRouter component 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      exact={true}
      path="/"
    >
      <Home />
    </Route>
    <Connect(Component)
      path="/register"
    >
      <Register />
    </Connect(Component)>
    <Connect(Component)
      path="/login"
    >
      <Login />
    </Connect(Component)>
    <Connect(Component)
      path="/secret"
    >
      <Secret />
    </Connect(Component)>
  </Switch>
</BrowserRouter>
`;
